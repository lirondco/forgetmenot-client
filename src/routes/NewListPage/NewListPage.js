import React, { Component } from 'react'
import NewListForm from '../../components/NewListForm/NewListForm'
import { Section } from '../../components/Utils/Utils'

export default class NewListPage extends Component {
    static defaultProps = {
        history: {
            push: () => {}
        },
    }

    handlePostListSuccess = list => {
        const { history } = this.props
        history.push('/')
    }

    render() {
        return (
            <Section className = 'newListPage'>
                <NewListForm
                 onPostListSuccess={this.handlePostListSuccess}
                />
            </Section>
        )
    }
}