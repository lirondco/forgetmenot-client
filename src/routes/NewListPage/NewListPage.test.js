import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewListPage from './NewListPage'

describe(`NewListPage component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<NewListPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})