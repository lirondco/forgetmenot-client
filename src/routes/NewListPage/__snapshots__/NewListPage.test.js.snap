// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListPage component renders the complete form 1`] = `
<Section
  className="newListPage"
>
  <NewListForm
    onPostListSuccess={[Function]}
  />
</Section>
`;
